code/modules/fish/fish_tank.dm

```diff
@@ -0,0 +1,685 @@
+	//Destruction check
+	if(cur_health <= 0)								//The tank is broken, destroy it
+		destroy()
+
+/obj/machinery/fishtank/proc/kill_fish(var/type = null)
+	//Check if we were passed a fish to kill, otherwise kill a random one
+	if(type)
+		fish_list.Remove(type)						//Kill a fish of the specified type
+	else
+		fish_list.Remove(pick(fish_list))			//Kill a random fish
+
+/obj/machinery/fishtank/proc/add_fish(var/type)
+	//Check if we were passed a fish type
+	fish_list.Add("[type]")						//Add a fish of the specified type
+	//Announce the new fish
+	visible_message("A new [type] has hatched in \the [src]!")
+
+/obj/machinery/fishtank/proc/select_egg_type()
+	var/fish = pick(fish_list)						//Select a fish from the fish in the tank
+	if(prob(25))									//25% chance to be a dud (blank) egg
+		fish = "dud"
+	var/obj/item/fish_eggs/egg_path	= null			//Create empty variable to receive the egg_path
+	egg_path = fish_eggs_list[fish]					//Locate the corresponding path from fish_eggs_list that matches the fish
+	if(!egg_path)									//The fish wasn't located in the fish_eggs_list, potentially due to a typo, so return a dud egg
+		return /obj/item/fish_eggs
+	else											//The fish was located in the fish_eggs_list, so return the proper egg
+		return egg_path
+
+/obj/machinery/fishtank/proc/harvest_eggs(var/mob/user)
+	if(!egg_list.len)									//Can't harvest non-existant eggs
+		return
+
+	for(var/i in egg_list.len)						//Loop until you've harvested all the eggs
+		var/obj/item/fish_eggs/egg = pick(egg_list)	//Select an egg at random
+		egg = new egg(get_turf(user))				//Spawn the egg at the user's feet
+		egg_list.Remove(egg)						//Remove the egg from the egg_list
+
+	egg_list.Cut()									//Destroy any excess eggs, clearing the egg_list
+
+/obj/machinery/fishtank/proc/harvest_fish(var/mob/user)
+	if(!fish_list.len)									//Can't catch non-existant fish!
+		to_chat(user, "There are no fish in \the [src] to catch!")
+		return
+
+	var/caught_fish = input("Select a fish to catch.", "Fishing") as null|anything in fish_list		//Select a fish from the tank
+	if(caught_fish)
+		var/dead_fish = null
+		dead_fish = fish_items_list[caught_fish]	//Locate the appropriate fish_item for the caught fish
+		if(!dead_fish)								//No fish_item found, possibly due to typo or not being listed. Do nothing.
+			return
+		kill_fish(caught_fish)						//Kill the caught fish from the tank
+		user.visible_message("[user.name] harvests \a [caught_fish] from \the [src].", "You scoop \a [caught_fish] out of \the [src].")
+		new dead_fish(get_turf(user))				//Spawn the appropriate fish_item at the user's feet.
+
+/obj/machinery/fishtank/proc/destroy(var/deconstruct = FALSE)
+	if(!deconstruct)
+		to_chat(world, "Brutally")														//Check if we are deconstructing or breaking the tank
+		for(var/i in  shard_count)														//Produce the appropriate number of glass shards
+			var/obj/item/weapon/shard/S = new /obj/item/weapon/shard
+			S.forceMove(get_turf(src))
```

Comment:
Note to reviewers : this doesn't work, the shards don't actually appear

Reply:
Just do new /obj/item/weapon/shard(get_turf(src))

Reply:
This was the it was before and it didn't work either.